build/php.m4

```diff
@@ -2217,6 +2217,17 @@ crypt_r("passwd", "hash", &buffer);
     fi
     ])
 
+    if test "$php_cv_crypt_r_style" = "none"; then
+      AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[
+#include <stdlib.h>
+#include <unistd.h>
+]],[[
+struct crypt_data buffer;
+crypt_r("passwd", "hash", &buffer);
+]])],[php_cv_crypt_r_style=struct_crypt_data_bsd],[])
```

Comment:
```suggestion
]])],[php_cv_crypt_r_style=struct_crypt_data],[])
```
I think. Though I think it would be better to just add an `#if HAVE_UNISTD_H #include <unistd.> #endif` to the main check above, instead of doing a separate one. It's not like including an extra header should hurt :)

Reply:
there is no crypt.h header on FreeBSD

Reply:
I see, in that case I'd still suggest to merge the checks and put the crypt.h include behind `#if HAVE_CRYPT_H` as well.

As implemented now, I would expect actual usage to fail because neither CRYPT_R_STRUCT_CRYPT_DATA or CRYPT_R_CRYPTD gets set. It doesn't matter right now because we never actually end up using system crypt_r right now, but would be a problem if we did.

Reply:
Yes right I knew for crypt_r not used it was just to pass config step :-)

Reply:
Your current variant also works though, maybe it's cleaner this way :)

Reply:
The current version works better maybe in the longer run. What do you prefer ?

Reply:
Yeah, let's stick with your current version.